st2tests/st2tests/action_aliases.py

```diff
@@ -0,0 +1,125 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import inspect
+
+from unittest2 import TestCase
+
+from st2common.content.loader import ContentPackLoader
+from st2common.exceptions.content import ParseException
+from st2common.bootstrap.aliasesregistrar import AliasesRegistrar
+from st2common.models.utils.action_alias_utils import extract_parameters_for_action_alias_db
+from st2common.models.utils.action_alias_utils import extract_parameters
+
+__all__ = [
+    'BaseActionAliasTestCase'
+]
+
+
+class BaseActionAliasTestCase(TestCase):
+    """
+    Base class for testing action aliases.
+    """
+
+    action_alias_name = None
+    action_alias_db = None
+
+    def setUp(self):
+        super(BaseActionAliasTestCase, self).setUp()
+
+        if not self.action_alias_name:
+            raise ValueError('"action_alias_name" class attribute needs to be provided')
+
+        self.action_alias_db = self._get_action_alias_db_by_name(name=self.action_alias_name)
+
+    def assertCommandMatchesFormatString(self, format_string, command):
+        """
+        Assert that the provided command matches the provided format string.
+        """
+        try:
+            extract_parameters_for_action_alias_db(action_alias_db=self.action_alias_db,
+                                                   format_str=format_string,
+                                                   param_stream=command)
+        except ParseException as e:
+            raise AssertionError(str(e))
+
+    def assertCommandMatchesSingleFormatString(self, format_strings, command):
```

Comment:
Thanks @Kami, 

I think I could live just the addition of `assertCommandMatchesSingleFormatString` if it asserted if there was no matches. Which is really what `assertCommandMatchesFormatString` is attempting to cover.

This is just to allow pack authors to test a set of aliases as whole instead of individually, thus ensuring we don't have an overlap (or incorrect ordering) of different representations.